extensions/control-plane/provision/provision-bigquery/src/main/java/org/eclipse/edc/connector/provision/gcp/BigQueryProvisioner.java

```diff
@@ -0,0 +1,177 @@
+            if (serviceAccount != null) {
+                token = iamService.createAccessToken(serviceAccount);
+            } else {
+                serviceAccount = ADC_SERVICE_ACCOUNT;
+                token = iamService.createDefaultAccessToken();
+            }
+            monitor.info("BigQuery Provisioner token ready");
+
+            var resource = getProvisionedResource(resourceDefinition, resourceName, tableName, serviceAccount);
+            var response = ProvisionResponse.Builder.newInstance().resource(resource).secretToken(token).build();
+            return CompletableFuture.completedFuture(StatusResult.success(response));
+        } catch (GcpException | IOException exception) {
+            return completedFuture(StatusResult.failure(ResponseStatus.FATAL_ERROR, exception.toString()));
+        }
+    }
+
+    private BigQueryProvisionedResource getProvisionedResource(BigQueryResourceDefinition resourceDefinition, String resourceName, String table, GcpServiceAccount serviceAccount) {
+        String serviceAccountName = null;
+        if (serviceAccount != null) {
+            serviceAccountName = serviceAccount.getName();
+        }
+
+        return BigQueryProvisionedResource.Builder.newInstance()
+            .properties(resourceDefinition.getProperties())
+            .id(resourceDefinition.getId())
+            .resourceDefinitionId(resourceDefinition.getId())
+            .transferProcessId(resourceDefinition.getTransferProcessId())
+            .resourceName(resourceName)
+            .project(resourceDefinition.getProject())
+            .dataset(resourceDefinition.getDataset())
+            .table(table)
+            .serviceAccountName(serviceAccountName)
+            .hasToken(true).build();
+    }
+
+    @Override
+    public CompletableFuture<StatusResult<DeprovisionedResource>> deprovision(
+            BigQueryProvisionedResource provisionedResource, Policy policy) {
+        return CompletableFuture.completedFuture(StatusResult.success(
+            DeprovisionedResource.Builder.newInstance()
+                .provisionedResourceId(provisionedResource.getId()).build()));
+    }
+
+    private String getServiceAccountName(BigQueryResourceDefinition resourceDefinition) {
+        if (resourceDefinition.getServiceAccountName() != null) {
+            // TODO verify service account name from resource definition before returning.
+            return resourceDefinition.getServiceAccountName();
+        }
+
+        return gcpConfiguration.serviceAccountName();
+    }
+
+    private BigQueryTarget getTarget(BigQueryResourceDefinition resourceDefinition) {
+        var project = resourceDefinition.getProject();
+        var dataset = resourceDefinition.getDataset();
+        var table = resourceDefinition.getTable();
+
+        if (project == null) {
+            project = gcpConfiguration.projectId();
+        }
+
+        return new BigQueryTarget(project, dataset, table);
+    }
+
+    public static class Builder {
```

Comment:
this builder is not really needed as it's relying on a single newInstance method + build, just use the constructor instead

Reply:
Done